test/hotspot/jtreg/compiler/c2/aarch64/TestTrampoline.java

```diff
@@ -0,0 +1,108 @@
+/*
+ * Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ *
+ */
+
+package compiler.c2.aarch64;
+
+import java.util.ArrayList;
+import java.util.Iterator;
+import jdk.test.lib.process.OutputAnalyzer;
+import jdk.test.lib.process.ProcessTools;
+
+/**
+ * @test TestTrampoline
+ * @summary Checks that trampolines to runtime code are not generated if they are not needed.
+ * @bug 8285487
+ * @library /test/lib
+ *
+ * @requires vm.flagless
+ * @requires os.arch=="aarch64"
+ * @requires vm.debug == false
+ * @requires vm.compiler2.enabled
+ *
+ * @run driver compiler.c2.aarch64.TestTrampoline
+ */
+
+public class TestTrampoline {
+    private final static int ITERATIONS_TO_HEAT_LOOP = 20_000;
+
+    public static void main(String[] args) throws Exception {
+        String testClassName = TestTrampoline.Test.class.getName();
+        ArrayList<String> command = new ArrayList<String>();
+        command.add("-XX:+UnlockDiagnosticVMOptions");
+        command.add("-Xbatch");
+        command.add("-XX:CompileCommand=print," + testClassName + "::" + "test");
+        // As the non-nmethod segment is put between other two segments,
+        // ReservedCodeCacheSize=240M guarantees no trampolines for runtime calls to the non-nmethod segment.
+        command.add("-XX:ReservedCodeCacheSize=240M");
```

Comment:
To reduce memory load on the test machines, I would reduce the ReservedCodeCacheSize to 140M.

Minor stylistic comment: let us remove empty lines 59, 61, 63, 65, 67 below.

Reply:
Reduced it to 130M.